strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,254 @@
+    e_x = []
+    e_y = []
+
+    for e in graph.edges():
+
+        start_x, start_y = l[e[0]]
+        end_x, end_y = l[e[1]]
+
+        e_x.append(start_x)
+        e_x.append(end_x)
+
+        e_y.append(start_y)
+        e_y.append(end_y)
+
+        e_x.append(None)
+        e_y.append(None)
+
+    return {"x": e_x, "y": e_y}
+
+
+GREEN = "#3e9651"
+RED = "#cc2529"
+LIGHT_GREY = "#CDCDCD"
+
+graph_node_colour = GREEN
+graph_edge_colour = LIGHT_GREY
+subgraph_node_colour = RED
+subgraph_edge_colour = RED
+
+graph_node_size = 14
+subgraph_node_size = 16
+
+
+def plot(graph: nx.Graph, subgraph: Optional[list] = None, size: float = 500) -> None:
+    """Creates a plotly plot of the input graph.
+
+    This function can plot just the input graph or the graph with a specified subgraph highlighted.
+
+    **Example usage**:
+
+    >>> graph = nx.complete_graph(10)
+    >>> fig = plot(graph, [0, 1, 2, 3])
+    >>> fig.show()
+
+    .. image:: ../../_static/complete_graph.png
+       :width: 40%
+       :align: center
+       :target: javascript:void(0);
+
+    Args:
+        graph (nx.Graph): input graph
+        subgraph (list): optional list of nodes comprising the subgraph to highlight
+        size (float): size of the plot in pixels, rendered in a square layout
+
+    Returns:
+         Figure: Plotly figure for graph and optionally highlighted subgraph
+    """
+    try:
+        import plotly.graph_objects as go
+    except ImportError:
+        raise ImportError("Plotly required for using plot()")
+    except RuntimeError:
+        print("Plotly unable to open display")
+        raise
+
+    s = graph.subgraph(subgraph)
+    l = nx.kamada_kawai_layout(graph)
+
+    g_nodes = go.Scatter(
+        **_node_coords(graph, l),
+        mode="markers",
+        hoverinfo="text",
+        marker=dict(color=graph_node_colour, size=graph_node_size, line_width=2),
+    )
+
+    g_edges = go.Scatter(
+        **_edge_coords(graph, l),
+        line=dict(width=1, color=graph_edge_colour),
+        hoverinfo="none",
+        mode="lines",
+    )
+
+    g_nodes.text = [str(i) for i in graph.nodes()]
+
+    layout = go.Layout(
+        showlegend=False,
+        hovermode="closest",
+        xaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+        yaxis=dict(showgrid=False, zeroline=False, showticklabels=False),
+        margin=dict(b=0, l=0, r=0, t=25),
+        height=size,
+        width=size,
+        plot_bgcolor="#ffffff",
+    )
+
+    if subgraph:
+
+        s_edges = go.Scatter(
+            **_edge_coords(s, l),
+            line=dict(width=2, color=subgraph_edge_colour),
+            hoverinfo="none",
+            mode="lines",
+        )
+
+        s_nodes = go.Scatter(
+            **_node_coords(s, l),
+            mode="markers",
+            hoverinfo="text",
+            marker=dict(color=subgraph_node_colour, size=subgraph_node_size, line_width=2),
+        )
+
+        s_nodes.text = [str(i) for i in s.nodes()]
```

Comment:
One thing I noticed is that when the user hovers over a subgraph node, the colour can change between red and green which looks like a bug. I think we can fix this by having `g_nodes` only be the nodes in the complement graph if a subgraph is specified.